trl/trainer/gkd_trainer.py

```diff
@@ -100,6 +100,7 @@ def __init__(
             do_sample=True,
             top_k=0,
             use_cache=False if args.gradient_checkpointing else True,
+            eos_token_id=args.eos_token_id,
```

Comment:
this can't be `eos_token_id=model_config.eos_token_id` ?

or better yet, `self.generation_config = model.generation_config`
then override temperature etc.

Reply:
Ah good point, that could well work since I see these IDs are provided in the model config directly: https://huggingface.co/meta-llama/Llama-3.1-8B-Instruct/blob/0e9e39f249a16976918f6564b8830bc894c89659/config.json#L8